### DISCUSSION TITLE
Looking to hire someone to fix problems with CEFSharp.DOM and add some customization

### DISCUSSION THREAD

**Discussion by gunpal5** [cefsharp/CefSharp.Dom#38] [General]
Hello,

We are looking for someone to customize Cefsharp.DOM to our needs and fix the bugs in the library. Basically we want similar level of DOM accessibility and web automation capabilities like GeckoFX has. 

Please let us know if there's someone who can help us achieve such thing.

Regards

**Comment by amaitland** [comment]
Hi @gunpal5,

Thanks for your interest in the project 👍 I'm the maintainer/owner of this project and also the `CefSharp` maintainer.

> fix the bugs in the library

I see you commented on https://github.com/cefsharp/CefSharp.Dom/discussions/34 was that the issue you were hoping for help with or are there others?

An assessment would need to be made to determine if it's fixable or a workaround investigated. I vaguely remember that frames were being attached then immedetly detached, so the issue might actually need to be resolved in [CEF](https://bitbucket.org/chromiumembedded/cef/) first before `OOPIF` support can be added.

> Basically we want similar level of DOM accessibility and web automation capabilities like GeckoFX has.

As I've never used `GeckoFx` I'm not familiar with how you access the `DOM`, can you provide some additional information? Usage examples? etc

To be clear from the start the `Chrome DevTools Protocol` is `asynchronous`, if you are looking for a `synchronous API` then unfortunately that's not something I'll be adding to this library.

If you require help migrating your `GeckoFx` to `CefSharp.Dom` then that's something I can help you with.

My rate is $90USD per hour with a two hour minimum payable via [GitHub Sponsors](https://github.com/sponsors/amaitland) (you can make one time payments via the interface). 

Thanks,
Alex

**Reply by gunpal5** [reply]
@amaitland Thank you for your reply. Like you said CEF has issues with the `OOPIF`. So, I think, best we leave it for now. 

I trying your other DOM library for WebView2 on Github. It seems to be working better than CEFSharp.DOM. If it resolved the issues with our software, then we will join your sponsorship program to support your efforts to maintain the library.

Thank you,
Gunpal Jain
